Pod/Classes/PINRemoteImageManager.m

```diff
@@ -831,7 +831,10 @@ - (PINDataTaskOperation *)downloadDataWithURL:(NSURL *)url
                 NSData *data = task.progressImage.data;
             [strongSelf unlock];
             
-            completion(data, error);
+            //if task has been removed, no point in calling completion
```

Comment:
We could also call the completion block with a specific "task remove" error. I'm not which approach is more idiomatic, but it seems friendlier to always call a completion block if one is provided so that the caller can take care of any housekeeping.